File: code/backend/app/schemas/analytics.py
from pydantic import BaseModel


class AnalyticsOverview(BaseModel):
    total_equipment: int
    available_equipment: int
    total_bookings: int
    confirmed_bookings: int
    active_transactions: int
    overdue_transactions: int
    active_defaulters: int


class EquipmentAnalytics(BaseModel):
    equipment_id: int
    name: str
    category: str
    total_bookings: int
    total_issues: int
    utilization_rate: float


class BookingAnalytics(BaseModel):
    total: int
    requested: int
    confirmed: int
    waitlisted: int
    cancelled: int
    no_show: int
    completed: int


class UsageTrend(BaseModel):
    date: str
    bookings: int
    issues: int
    returns: int


class PeakHour(BaseModel):
    hour: int
    booking_count: int


class DamageStats(BaseModel):
    total_returns: int
    damaged_returns: int
    damage_rate: float
    recent_damage_reports: list[dict]

File: code/web_dashboard/src/types/analytics.ts
// All shapes mirror backend app/schemas/analytics.py exactly.

export interface AnalyticsOverview {
  total_equipment: number;
  available_equipment: number;
  total_bookings: number;
  confirmed_bookings: number;
  active_transactions: number;
  overdue_transactions: number;
  active_defaulters: number;
}

export interface EquipmentAnalytics {
  equipment_id: number;
  name: string;
  category: string;
  total_bookings: number;
  total_issues: number;
  utilization_rate: number;
}

// Single status-count object — NOT a time series.
export interface BookingAnalytics {
  total: number;
  requested: number;
  confirmed: number;
  waitlisted: number;
  cancelled: number;
  no_show: number;
  completed: number;
}

export interface UsageTrend {
  date: string;
  bookings: number;
  issues: number;
  returns: number;
}

export interface PeakHour {
  hour: number;
  booking_count: number;
}

export interface DamageReport {
  transaction_id: number;
  equipment_id: number;
  damage_report: string;
  returned_at: string;
}

export interface DamageStats {
  total_returns: number;
  damaged_returns: number;
  damage_rate: number;
  recent_damage_reports: DamageReport[];
}

export interface FairnessConfig {
  id: number;
  daily_usage_cap: number;
  recency_threshold_days: number;
  recency_weight: number;
  updated_at: string;
  updated_by: number | null;
}
